### PR TITLE
Add GRPC Endpoints

### PR DESCRIPTION
**Pull request by KhaledHegazy222**

### CODE REVIEW

**Review by AbdelrahmanElawady — DISMISSED** [review]
LGTM

File: server/server.go
Comment (tareknaser) [inline server/server.go:1]: Consider relocating the`server/` directory to `internal/server/` directory, to align with Go best practices.

File: cmd/server.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	Short: "Run the GRPC Server",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		grpcServer := server.NewServer()
+		err := grpcServer.Start(grpcServerPort)
+		if err != nil {
+			log.Fatalf("Unexpected Error: %v", err)
+		}
+	},
+}
```
Comment (tareknaser) [inline cmd/server.go:28]: Consider placing `serverCmd` variable declaration above `init()` function to improve readability.

File: cmd/server.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+var grpcServerPort int16
+
+func init() {
+	serverCmd.PersistentFlags().Int16VarP(&grpcServerPort, "port", "p", 8080, "grpc server port")
+	serverCmd.MarkPersistentFlagRequired("port")
+}
+
+var serverCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Run the GRPC Server",
```
Comment (tareknaser) [inline cmd/server.go:19]:
```suggestion
	Short: "Run the gRPC Server",
```

**Review by tareknaser — APPROVED** [review]
This is already in good shape, just a few minor suggestions.

**Comment by KhaledHegazy222** [comment]
Closing this pull request because we plan to merge the backend API code with the infrastructure software in the same application. Therefore, we won't require any gRPC communication.

**Pull request closed without merging**